src/context/Authentication/index.js

```diff
@@ -13,60 +16,90 @@ import { getCurrentUser } from '../../hooks/Api/Users/Get';
 const AuthContext = React.createContext();
 
 const AuthProvider = ({ children }) => {
-  const [isAuthenticated, setIsAuthenticated] = useState(false);
+  const [isAuthenticated, setIsAuthenticated] = useState();
   const [cognitoUser, setCognitoUser] = useState();
   // customUserData refers to the attributes we have saved in dynamo
-  const [customUserData, setCustomUserData] = useState();
+  const [customUserData, setCustomUserData] = useState({});
   const [token, setToken] = useState();
-  const [userId, setUserId] = useState();
   const [tempEmail, setTempEmail] = useState();
+  // Replaces the basic useState for the userId
+  // This way, all the updates to this value will also be saved to localhost
+  const [userId, setUserId] = useLocalStorageUser();
 
-  // Check if the user is already signed in in the beginning
+  // On page load
   useEffect(() => {
     if (typeof window !== `undefined`) {
+      // Check if the user is already signed in
       import(/* webpackChunkName: "Amplify" */ '@aws-amplify/auth').then(
         ({ default: Amplify }) => {
           Amplify.currentAuthenticatedUser()
-            .then(user => setCognitoUser(user)) // set user in context (global state)
-            .catch(() => setCognitoUser(null)); //error is thrown if user is not authenticated
+            .then(user => {
+              if (user) {
+                setCognitoUser(user);
+                setIsAuthenticated(true);
+              }
+            }) // set user in context (global state)
+            .catch(() => {
+              //error is thrown if user is not authenticated
+              setIsAuthenticated(false);
+            });
         }
       );
+
+      // Check for URL param for user ID
+      const params = querystring.parse(window.location.search);
+      // If params, set that user as the userId and via localStorage
+      if (!userId && params.userId) {
+        setUserId(params.userId);
```

Comment:
@Valioesi and I talked about this. He said to have it keep the current userId. I didn't fully get why. But I kinda see why logging out when seeing a new userId and identifying a new user makes sense.

Reply:
Ah, okay. @timurc we had a similar discussion once: what should happen if there is already an existing session when someone comes to a page like self scan with a userId in the params? Back then we decided that the user id in the params should not be regarded and the session should be used instead.

That's why I figured we would just do it the same way here. But I guess it's kind of a different situation, because we don't have real session. 

I am fine with changing it to the new user id in this case. I think it makes sense. :)